bcgov/cas-cif
# Chore/1333 replace terraform cloud

Addresses #1333 - [Project link](https://github.com/orgs/bcgov/projects/123/views/2?filterQuery=1333&pane=issue&itemId=49863829)
Adds Terraform scripts and Helm charts to run them. Resource states migrated from Terraform Cloud to a Google Cloud Storage bucket.

## Changes 🚧

- Added Terraform module, imported from cas-shelf
- Added Helm templates to handle:
  - Terraform module loading to configMaps
  - OpenShift service account for use by Terraform
  - Terraform apply job

## To test 🔬

- The Helm charts are built into the automation around deploys via ShipIt. Once a deploy has been run, you can check the job `Terraform-apply` in OpenShift to see if the run was successful.

## Notes📝 
- 🔒 Terraform Cloud state has been locked to enable migration of state to state buckets
- The script has been run and resources have been transitioned from Terraform Cloud to GCP buckets holding state for each namespace (cif-dev, cif-test, cif-prod).  
- Resources have been migrated rather than recreated, only the state "owning" the resource changes. 
- The resources `kubernetes_secret.secret_sa` have had Terraform changes, but only the addition of a default value (consistent with our use of them) due to a new provider version. 
- The script used for migration (/chart/cas-cif/terraform/tf-migration.sh) is kept here for history. It will be removed by https://github.com/bcgov/cas-pipeline/issues/83
